### ISSUE TITLE
Unable to run `pg_dump` when following migrating between projects documentation

### ISSUE THREAD

**Issue by peterp** [supabase/supabase#1856]
# Improve documentation

## Link 

https://supabase.io/docs/guides/database#migrating-between-projects

## Describe the problem

I'm trying to add restoration support for Supabase in https://snaplet.dev, but whilst trying to dump the database I get the following error:

```terminal
$ pg_dump -h db.PROJECT_REF.supabase.co -U postgres --clean --schema-only M> supabase_schema.sql

pg_dump: error: query failed: ERROR:  permission denied for table migrations
pg_dump: error: query was: LOCK TABLE storage.migrations IN ACCESS SHARE MODE
```

I can exclude the `storage` schema, but it would be nice to have a completely functioning restoration.

**Comment by inian** [comment]
Hi @peterp, we encountered this issue internally as well. For now, you can exclude the `storage.migrations` table in your pg_dump. The rest of the tables in storage schema should be fine. We will roll out the fix to be able to dump the migrations table as well soon.

**Comment by inian** [comment]
Hi @peterp, this has been fixed for new projects. If you can send your project ref here or to beta@supabase.io, I will add the patch to your database too.

**Comment by JasonChiu-dev** [comment]
I have the same situation as well.

When I follow the guidance from the document: https://supabase.io/docs/guides/database#migrating-between-projects
and run the following command 

$ pg_dump -h db.PROJECT_REF.supabase.co -U postgres --clean --schema-only  > supabase_schema.sql

the following errors occurred:

pg_dump: error: query failed: ERROR:  permission denied for table migrations
pg_dump: error: query was: LOCK TABLE storage.migrations IN ACCESS SHARE MODE

How can I solve this problem?

**Comment by JasonChiu-dev** [comment]
correct the error message:

pg_dump: error: query failed: ERROR:  permission denied for table identities
pg_dump: error: query was: LOCK TABLE auth.identities IN ACCESS SHARE MODE

**Comment by bnjmnt4n** [comment]
@JasonChiu-dev see the following comment: https://github.com/supabase/supabase/discussions/3464#discussioncomment-1616225:

```sql
grant all on auth.identities to postgres, dashboard_user;
```

**Comment by JasonChiu-dev** [comment]
> @JasonChiu-dev see the following comment: [#3464 (reply in thread)](https://github.com/supabase/supabase/discussions/3464#discussioncomment-1616225):
> 
> ```sql
> grant all on auth.identities to postgres, dashboard_user;
> ```

Cool! It really works. 
Is there any adjusment I need to do after I run "grant all on auth.identities to postgres, dashboard_user;"?

**Comment by JasonChiu-dev** [comment]
Hi Benjamin,

Thank you so much for your help.

Jason

Benjamin Tan ***@***.***> 於 2021年11月10日 週三 13:32 寫道：

**Comment by soedirgo** [comment]
@JasonChiu-dev there should be no adjustment needed - it's a known issue which will be fixed soon 🙏

**Comment by JasonChiu-dev** [comment]
Thank you.

Jason


Bobbie Soedirgo ***@***.***> 於 2021年11月10日 週三 17:58 寫道：

**Comment by bigonha** [comment]
I'm still having the same issue.

**Comment by soedirgo** [comment]
@bigonha can you try following the updated migration instructions here: https://supabase.com/docs/guides/database#migrating-between-projects?

**Comment by TomasHubelbauer** [comment]
For anyone who's looking to just dump/backup all their Supabase database data to have a local backup and do not care about the `auth` and `realtime` schema contents:

```
pg_dump -d postgresql://postgres@db.{project}.supabase.co:5432/postgres -N auth -N realtime > dump.sql
```

You'll be asked for the password you set during the setup of your Supabase project. If you don't have `pg_dump`, do not want to install full Postgres locally and are a macOS user, look into https://postgresapp.com/ (https://github.com/PostgresApp/PostgresApp) - you can skip the database initialization step and just add the CLI tools to your command line.

**Comment by relferreira** [comment]
For anyone facing this problem, what worked for me was execute this SQL in the editor:
```
ALTER ROLE postgres SUPERUSER
```

as described in this tutorial:
https://supabase.com/docs/guides/database#migrating-between-projects

**Comment by princefishthrower** [comment]
@relferreira - but then once the dump is finished, *it's very important to then switch back*:

```sql
ALTER ROLE postgres NOSUPERUSER
```

@TomasHubelbauer adding `-N auth -N realtime` didn't solve the original issue that @peterp reported for me, i.e.:

```
pg_dump: error: query failed: ERROR:  permission denied for table migrations
pg_dump: error: query was: LOCK TABLE supabase_functions.migrations IN ACCESS SHARE MODE
```

In this case I also had to add `-N supabase_functions` (though probably not ideal for when you want to fully inspect a dump) I raised another issue here that is related: https://github.com/supabase/supabase/issues/7865 it's unclear to me why `SUPERUSER` is needed at all, I don't remember that being a part of the dump workflow when I've worked with Postgres previously...

**Comment by jordiup** [comment]
> ALTER ROLE postgres NOSUPERUSER

Cheers @princefishthrower was looking for this answer 🚀

**Comment by mratose** [comment]
GRANT USAGE ON SCHEMA schema TO user;

**Comment by Strajk** [comment]
got the same error while trying to set up backups with [snapshooter.com](https://snapshooter.com/)

![image](https://user-images.githubusercontent.com/697301/228944201-d08a64b3-7a70-409a-8aec-22989dffff7c.png)

![Screen 2023-03-30 at 21 33 02](https://user-images.githubusercontent.com/697301/228944461-d7da2bee-5ca1-4fc9-8224-3e3c9c94d73c.png)

**Comment by madebyfabian** [comment]
In my case, adding `-N supabase_functions` was enough.
